### PR TITLE
Implement vanilla gen 1.16.100

### PR DESCRIPTION
**Pull request by KCodeYT**

### CODE REVIEW

File: gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java [hunk 1]
```diff
@@ -1748,6 +1748,8 @@ public boolean shouldBeSeen(EntityPlayer player) {
     }
 
     public boolean knowsChunk(ChunkAdapter adapter) {
+        if(adapter == null)
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java:1751]: In which case can this be null?

File: gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/VanillaGeneratorImpl.java [hunk 2]
```diff
@@ -202,8 +204,10 @@ public VanillaGeneratorImpl(World world, GeneratorContext context) {
             String data = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
 
             // Replace the port
-            data = data.replace("server-port=19132", "server-port=0");
-            data = data.replace("server-portv6=19133", "server-portv6=0");
+            String portLine = Arrays.stream(data.split("\n")).filter(line -> line.contains("server-port=")).collect(Collectors.joining());
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/VanillaGeneratorImpl.java:211]: This is not safe to use. Nothing checks if the port really is useable or not

File: gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java [hunk 3]
```diff
@@ -9,43 +9,16 @@
 
 import com.google.common.util.concurrent.ListenableScheduledFuture;
 import io.gomint.crypto.Processor;
-import io.gomint.jraknet.ClientSocket;
-import io.gomint.jraknet.Connection;
-import io.gomint.jraknet.EncapsulatedPacket;
-import io.gomint.jraknet.PacketBuffer;
-import io.gomint.jraknet.PacketReliability;
-import io.gomint.jraknet.SocketEvent;
+import io.gomint.jraknet.*;
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java]: No asteriks pls

File: gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java [hunk 4]
```diff
@@ -9,43 +9,16 @@
...
-import io.gomint.server.network.packet.PacketLogin;
-import io.gomint.server.network.packet.PacketMovePlayer;
-import io.gomint.server.network.packet.PacketPlayState;
-import io.gomint.server.network.packet.PacketRequestChunkRadius;
-import io.gomint.server.network.packet.PacketResourcePackResponse;
-import io.gomint.server.network.packet.PacketSetLocalPlayerAsInitialized;
-import io.gomint.server.network.packet.PacketStartGame;
-import io.gomint.server.network.packet.PacketWorldChunk;
+import io.gomint.server.network.*;
+import io.gomint.server.network.packet.*;
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java]: No asteriks pls

File: gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java [hunk 5]
```diff
@@ -64,26 +37,14 @@
 import java.nio.ByteOrder;
 import java.security.PublicKey;
 import java.time.Duration;
-import java.util.ArrayList;
-import java.util.Base64;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Queue;
-import java.util.UUID;
+import java.util.*;
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java]: No asteriks pls

File: gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java [hunk 6]
```diff
@@ -64,26 +37,14 @@
...
 
-import static io.gomint.server.network.Protocol.PACKET_DISCONNECT;
-import static io.gomint.server.network.Protocol.PACKET_ENCRYPTION_REQUEST;
-import static io.gomint.server.network.Protocol.PACKET_MOVE_PLAYER;
-import static io.gomint.server.network.Protocol.PACKET_PLAY_STATE;
-import static io.gomint.server.network.Protocol.PACKET_RESOURCEPACK_INFO;
-import static io.gomint.server.network.Protocol.PACKET_RESOURCEPACK_STACK;
-import static io.gomint.server.network.Protocol.PACKET_START_GAME;
-import static io.gomint.server.network.Protocol.PACKET_WORLD_CHUNK;
+import static io.gomint.server.network.Protocol.*;
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java]: No asteriks pls

File: gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java [hunk 7]
```diff
@@ -678,6 +623,7 @@ private void move(Location target) {
         movePlayer.setYaw(target.getYaw());
         movePlayer.setPitch(target.getPitch());
         movePlayer.setMode(PacketMovePlayer.MovePlayerMode.TELEPORT);
+        movePlayer.setTick(0);
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java:663]: Why a static value of 0 here? We could use the server tick here

**Review by geNAZt — CHANGES_REQUESTED** [review]

File: gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java (hunk 1 above)
Comment (KCodeYT) [reply gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java:1751]: I dont know, I got an error whilst testing

File: gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java (hunk 7 above)
Comment (KCodeYT) [reply gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java:663]: You cant send ticks as a client

Comment (KCodeYT) [reply gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/client/Client.java:663]: Only when you use the PlayerAuthInputPacket, but then we would have to rewrite how the bot is moving

File: gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/VanillaGeneratorImpl.java [hunk 8]
```diff
@@ -284,6 +288,17 @@ public VanillaGeneratorImpl(World world, GeneratorContext context) {
         }
     }
 
+    private int getRandomPort() {
+        return IntStream.range(50000, 60000).filter(port -> {
```
Comment (Janmm14) [inline gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/VanillaGeneratorImpl.java:292]: Better: We could just use a while where we grab a random number up to 10000 times for testing.

File: gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/VanillaGeneratorImpl.java [hunk 9]
```diff
@@ -284,6 +288,17 @@ public VanillaGeneratorImpl(World world, GeneratorContext context) {
...
+            try(DatagramSocket datagramSocket = new DatagramSocket()) {
+                datagramSocket.bind(new InetSocketAddress("127.0.0.1", port));
+                return true;
+            } catch(SocketException e) {
+                return false;
+            }
+        }).findFirst().orElseThrow();
```
Comment (Janmm14) [inline gomint-server/src/main/java/io/gomint/server/world/generator/vanilla/VanillaGeneratorImpl.java]: Let the system find a free port:
```suggestion
        try(DatagramSocket datagramSocket = new DatagramSocket(0)) {
            return datagramSocket.getLocalPort()
        } catch(SocketException ex) {
            throw new RuntimeException("Could not open socket to find next free port", ex);
        }
```

File: gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java (hunk 1 above)
Comment (geNAZt) [reply gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java:1751]: Ok i will check, i doubt this has something todo with your changes

**Review by geNAZt — APPROVED** [review]

**Pull request merged**